Blog: Add news about migration to Weblate (#158)
mumble-voip/mumble-www: hugo/content/blog/2021-03-20-weblate-translation/index.md

---
title: Switch to Weblate as Translation Platform
author: Jan Klass
date: 2021-05-10
categories:
  - News
---
The Mumble Client is available in many languages. For the translation process we recently moved from the Transifex online platform to Weblate. In this post we describe our translation history, reasoning, and current approach.

> Translations now happen in the [Mumble Client project on Hosted Weblate](https://hosted.weblate.org/projects/mumble/mumble-client/).

Reviewer: ```suggestion
Translations now happen in the [Mumble client project on Hosted Weblate](https://hosted.weblate.org/projects/mumble/mumble-client/).

```

<!--more-->

## History - From local to online

Our Mumble client is making use of the Qt library which provides various features, and amongst them a system for localizing.

In the source folder `src/mumble` we produce a `.ts` file for each language, which contains the source texts and corresponding translated text. These files could be edited manually with a text or XML editor, or a translation application.

To lower the barrier for potential translators we moved to [Transifex](https://www.transifex.com/), a web based translation platform in 2014. Using a web platform contributors and anyone else can access translation information with a web browser, which made it a lot easier to work with. It also adds a few collaborative features that can be used.

To integrate the third party service we implemented a bot ([MumbleTransifexBot](https://github.com/mumble-voip/MumbleTransifexBot)) that would pull translation updates and create pull requests on GitHub for us. Once established this significantly reduced the work necessary for maintaining this system and integrating translation changes.

## Reevaluation

> In the 6 years since we switched to Transifex, the translation platform changed significantly. Good alternatives appeared, and we took some discussion from a issue ticket as initiation for a reevaluation of our situation and opportunities.

Reviewer: ```suggestion
In the 6 years since we switched to Transifex, the translation platform changed significantly. Good alternatives appeared, and we took some discussion from an issue ticket as initiation for a reevaluation of our situation and opportunities.

```

## Migration to Weblate

The migration itself has mostly technical aspects to it, and a lot of related work. Deep-diving into the localization setup and aspects of our project also showed some inconsistencies and questions for the future, and potential for cleanup which we applied. The migration is/was tracked in [ticket #4727](https://github.com/mumble-voip/mumble/issues/4727).

Weblate provides integration to GitHub and pull request creation. Some advanced configuration scenarios allow for additional customization. Thanks to the integration we were able to phase out our integration and PR creation bot.

Interestingly enough even before finishing the migration and even with notices on Hosted Weblate to not translate yet we received contributions on there.

We have been using Weblate for several weeks now, and have received numerous contributions. Thank you to all existing and future translators!

The process is not as smooth as we would like it to be yet, but we are glad we made the transition already. And we hope existing and future translators like the Hosted Weblate platform.

## Weblate Translation Platform

> [Weblate](https://weblate.org/) is Open Source software. The hosted instance services not only us as FOSS but [other projects](https://hosted.weblate.org/projects/) as well.

Reviewer: ```suggestion
[Weblate](https://weblate.org/) is open source software. The hosted instance services not only us as FOSS but [other projects](https://hosted.weblate.org/projects/) as well.

```

Author: Didn’t we discuss this in some meeting? I thought we agreed to Open Source. But I guess it is lower case then?

Author: We have it as Open Source in the subheadline and open source in the About text on the landing page 🤔 😕

> We are very thankful for the free hosting they provide to us. If you are looking for a translation platform for other or [commercial projects (pricing)](https://weblate.org/en/hosting/), please consider using them. [Direct donations to them](https://weblate.org/en/donate/) are helping them too, and were explicitly mentioned for consideration after approving our project.

Reviewer: "for other or" - there seems to be something missing... "for other _FOSS_ or" perhaps?

## Our Translation Project on Hosted Weblate

We support numerous languages, and also have some languages that are not complete enough for productive use yet. If you are fluent in one of them, please consider contributing to [our translations](https://hosted.weblate.org/projects/mumble/mumble-client/).

![Weblate translation status by language](https://hosted.weblate.org/widgets/mumble/-/mumble-client/horizontal-auto.svg)

If you want to translate a language that we did not add yet, and are willing to commit to it, please create a request in [our issue tracker](https://github.com/mumble-voip/mumble/issues). We can add additional languages.

Have any feedback, positive or negative, praise or concerns? We would love to hear from you!

## Road to Mumble version 1.4

We recently landed blockers for 1.4, and are now in the process of preparing a last development snapshot, a feature freeze, and then plan to publish release candidates.

> We would love to release 1.4.0 with many, complete, and good translations. If you are a willing translator, please take this opportunity to focus on finalizing translations towards the upcoming release. It would be great if we had finalized translations within or in a month. Thank you very much.

Reviewer: ```suggestion
We would love to release 1.4.0 with many, complete, and good translations. If you are a willing translator, please take this opportunity to focus on finalizing translations towards the upcoming release. It would be great if we had finalized translations within about a month or two. Thank you very much.

```
